src/structures/Channel.js

```diff
@@ -97,6 +97,14 @@ class Channel extends Base {
     return this.client.channels.fetch(this.id, true, force);
   }
 
+  /**
+   * Indicates whether this channel is text-based.
+   * @returns {boolean}
+   */
+  isText() {
+    return ['text', 'dm', 'news'].includes(this.type);
```

Comment:
Because all text-based channels have `channel.messages`, this check can also be used to simplify code in the future, in case Discord adds more text-based channels.
```suggestion
    return Reflect.has(this, 'messages');
```

Reply:
Thanks for the idea :smile: